quickstart/backend/src/main/java/com/digitalasset/quickstart/service/AppInstallRequestsApiImpl.java

```diff
@@ -42,128 +51,199 @@ public AppInstallRequestsApiImpl(
-                    new quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Cancel(
-                            new quickstart_licensing.licensing.util.Metadata(appInstallRequestCancel.getMeta().getData())
-                    );
-
-            return ledger.exerciseAndGetResult(userParty, contract.contractId, choice, commandId)
-                    .thenApply(result -> {
-                        // Choice returns Unit, so we can return 200 OK
-                        return ResponseEntity.ok().<Void>build();
-                    }).exceptionally(ex -> {
-                        logger.error("Error cancelling AppInstallRequest", ex);
-                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-                    });
-        }).exceptionally(ex -> {
-            logger.error("Error fetching AppInstallRequest with contractId: {}", contractId, ex);
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
-        });
+        Span methodSpan = Span.current();
+        Context parentContext = Context.current();
+
+        methodSpan.addEvent("Starting cancelAppInstallRequest");
+        methodSpan.setAttribute("contractId", contractId);
+        methodSpan.setAttribute("commandId", commandId);
+        logger.info("cancelAppInstallRequest: contractId='{}', commandId='{}'", contractId, commandId);
+
+        return authenticatedPartyService.getPartyOrFail()
+                .thenCompose(userParty ->
+                        damlRepository.findAppInstallRequestById(contractId)
+                                .thenCompose(contract -> {
+                                    methodSpan.addEvent("Fetched contract, exercising AppInstallRequest_Cancel choice");
+
+                                    quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Cancel choice =
+                                            new quickstart_licensing.licensing.appinstall.AppInstallRequest.AppInstallRequest_Cancel(
+                                                    new quickstart_licensing.licensing.util.Metadata(appInstallRequestCancel.getMeta().getData())
+                                            );
+
+                                    return ledger.exerciseAndGetResult(userParty, contract.contractId, choice, commandId)
+                                            .thenApply(result -> {
+                                                methodSpan.addEvent("Choice exercised, returning 200 OK");
+                                                return ResponseEntity.ok().<Void>build();
+                                            });
+                                })
+                )
+                .whenComplete(
+                        completeWithin(parentContext, (res, ex) -> {
+                            if (ex == null) {
+                                logger.info("cancelAppInstallRequest: Success for contractId='{}'", contractId);
```

Comment:
With all of these logs, if you have the commandId, you should log the commandId.